src/sniffer.c

```diff
@@ -3685,6 +3704,19 @@ static int ProcessServerHello(int msgSz, const byte* input, int* sslBytes,
         }
     }
 
+    if (IsAtLeastTLSv1_3(session->sslServer->version)) {
+#ifndef WOLFSSL_TLS13
+        SetError(UNSUPPORTED_TLS_VER_STR, error, session, FATAL_ERROR_STATE);
```

Comment:
Hmm, those errors specifically would have to remain fatal since the application would otherwise continue decrypting and end up with `Bad application data`. To eliminate the hard coded stuff, we could add an arg to SetError to specify if `verboseErr`. Or we could just set `verboseErr` after calling that SetError:
```
SetError(UNSUPPORTED_TLS_VER_STR, error, session, FATAL_ERROR_STATE);
session->verboseErr=1;
```

The second would probably be the preference?

Reply:
I am okay with the `session->verboseErr = 1;` in those places where you want to return the original error string (not the generic fatal one).

Another possible solution is to not set the error string where you are handling `verboseErr` if the new `session->error` was already set?

Reply:
If `RemoveFatalSession()` is called, `session->error` will always have been set (since `SetError `would have been called prior to  `RemoveFatalSession()`. But the first suggestion sounds good. I'll push that up shortly